docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+
+## How to
+
+### Integrator
+
+In order to match the current EVM implementation from Reth, we want to run the same ethereum/tests as them. A list of all applicable tests can be found [here](https://github.com/paradigmxyz/reth/blob/main/testing/ef-tests/tests/tests.rs#L17). To start:
+
+- Check for an unassigned `epic` issue for a test.
+- Add it to  `crates/ef-testing/tests.rs` by using the `blockchain_tests` macro. The first argument should be the name of your test, the second is the folder to find the test. Please use the snake case name of the folder for the name of the test (e.g. `blockchain_tests!(st_bad_opcode, stBadOpcode)` for adding stBadOpcode).
+- Comment all the other lines except for the test you added and run `make ef-tests`.
+After a while (count 10-15 minutes) the test will end and you should have in your terminal's output a list of all the failing tests.
+- Start making issues. When raising issues, please try to match other `Integrator` issued on the style, and don't forget to add any raised issue to the `epic` issue's tasklist.
+
+### Debuggor
+
+Pick an issue from the available `Integrator` issues. Verify that the test fails (you can run a specific test by using `make target=your_test_name ef-test`). If it does, you can start debbugging it. The following documentation can be used:
+
+- [Test fillers](https://github.com/ethereum/tests/tree/develop/src/GeneralStateTestsFiller): used to generated the actual test. This can be used to understand what code the transaction executes.
+- [Execution specs](https://github.com/ethereum/execution-specs/tree/master): Ethereum execution spectifications written in Python. This can be used to compare the expected behaviour to the [Cairo 0 implementation](https://github.com/kkrt-labs/kakarot/tree/main/src).
+- [Yellow paper](https://ethereum.github.io/yellowpaper/paper.pdf): Ethereum format specification. This can be used to compare the expected behaviour to the [Cairo 0 implementation](https://github.com/kkrt-labs/kakarot/tree/main/src).
+
+#### Log opcodes
+
+The following section describes how to log the opcodes when running an ethereum/test:
+
+- Clone `https://github.com/kkrt-labs/kakarot-rpc.git`
+- Update the `Cargo.toml` file at the root of the repository, by replacing the blockifier import by `blockifier = { git = "https://github.com/jobez/blockifier.git", rev = "7f00407" }`.
+- In the file `./kakarot-rpc/lib/kakarot/src/kakarot/instructions.cairo`, add the following hint `%{print(ids.opcode, ids.ctx) %}` right after line 65. See the following [gist](https://gist.github.com/jobez/42941db9361d81778abd36309dfb60dc#file-instructions-cairo-L68-L70) for more details.
+- From the root of the `kakarot-rpc` repository, run the following: `cd lib/kakarot && STARKNET_NETWORK=starknet-devnet make build && cd ../.. && make RUST_BACKTRACE=1 dump-katana`
```

Comment:
shouldn't change a thing, but it's more `STARKNET_NETWORK=katana` than `starknet-devnet` currently

Reply:
The only issue right now is that compiling with "katana" will now run with the "--disable_hint_validation" flag, so then we need to add katana in the `compile_contract` function.

Reply:
i didn't understand

Reply:
So the reason why you need to set the `STARKNET_NETWORK` env var to `starknet-devnet`, is that this causes the compile scripts to compile the cairo 0 with the disable hints flag (see [here](https://github.com/kkrt-labs/kakarot/blob/main/scripts/utils/starknet.py#L268)). So if you want to update to `STARKNET_NETWORK=katana`, then we need to update the script to include katana, or this won't work.

Reply:
actually the only network so far accepting custom hints was starknet-devnet, if some others pop up, we can make this explicit in the script. But in any case, this will probably be updated with the new smal test sequencer